headers/types/common/common.h

```diff
@@ -818,46 +819,131 @@ struct mission_floors_forbidden {
 };
 ASSERT_SIZE(struct mission_floors_forbidden, 2);
 
+// Used throughout the process of displaying a mission
+#pragma pack(push, 1) // Align all members to a 1-byte boundary
+
+struct mission_details {
+    undefined field_1[8];
+    // 0x8: Pointer to "mission struct + 0x4" (new substruct?)
+    struct mission* mission_ptr;
+    undefined field_2[88];
+    // 0x60: pointer to mission_template struct
+    struct mission_template* template_ptr;
+};
+#pragma pack(pop) // Go back to the default packing behavior
+
+// mission_details is at least this big. unclear bigger it actually is.
+ASSERT_SIZE(struct mission_details, 104);
+
 // Information about a valid mission; a list of these structs is stored in and directly loaded from
-// RESCUE/rescue.bin. Used for validation as well as possibly for loading certain missions?
-struct mission_rescue_bin {
-    undefined field_0x0;
-    undefined field_0x1;
-    undefined field_0x2;
-    undefined field_0x3;
-    undefined field_0x4;
-    undefined field_0x5;
-    undefined field_0x6;
-    undefined field_0x7;
-    // 0x8: For legendary challenge missions, is used as a boolean to indicate whether or not
-    // accepting the mission unlocks its dungeon. For other missions, its purpose is unknown.
-    uint8_t unk_unlock_flag;
-    undefined field_0x9;
+// RESCUE/rescue.bin at 0x20, where 481 of these structs exist. These are used to select a string
+// from MISSION_STRING_IDS at random, to provide variance to mission flavor text.
+struct rescue_str_variant_group {
+    // 0x0: Index of MISSION_STRING_IDS the group starts from.
+    uint16_t starting_index;
+    // 0x2: Number of other MISSION_STRING_IDS entries the group contains. A random int on the range
+    // [0, group_size) is added to starting_index to produce the final MISSION_STRING_ID entry.
+    // This final index is reused to select an index in rescue_str_continuity_table
+    uint16_t group_size;
+};
+
```

Comment:
The style convention is to put the `ASSERT_SIZE` right after the struct definition without a blank line in between. Mainly just for compactness. Same thing applies to all the other `ASSERT_SIZE`s